src/rp2_common/pico_stdio_uart/stdio_uart.c

```diff
@@ -55,7 +57,7 @@ void stdout_uart_init() {
 }
 
 void stdin_uart_init() {
-#ifdef PICO_DEFAULT_UART_RX_PIN
+#if defined(uart_default) && defined(PICO_DEFAULT_UART_RX_PIN)
     bi_decl_if_func_used(bi_1pin_with_func(PICO_DEFAULT_UART_RX_PIN, GPIO_FUNC_UART));
 #if !defined(PICO_DEFAULT_UART_BAUD_RATE) || !defined(uart_default)
```

Comment:
AFAICT this is "inside" the outer `#if defined(uart_default) && ...` block, which suggests that the ` || !defined(uart_default)` here is redundant?

Reply:
fixed